instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/client/UnicastRefInstrumentation.java

```diff
@@ -46,28 +46,33 @@ public static void onEnter(
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
 
-      // TODO replace with client span check
-      if (!Java8BytecodeBridge.currentSpan().getSpanContext().isValid()) {
+      Context parentContext = Java8BytecodeBridge.currentContext();
+
+      // TODO replace with client span check - this is a bit different condition though, can we
+      // remove it?
+      if (!Java8BytecodeBridge.spanFromContext(parentContext).getSpanContext().isValid()) {
+        return;
+      }
```

Comment:
looks like tests pass without it. +1 to remove 🤷‍♂️

Reply:
But this is very different condition, no? I read it that we don't want to create top level spans for rmi calls. Nothing in our instrumenters does such check.